app/components/Article/index.js

```diff
@@ -30,21 +37,29 @@ const Article = React.createClass({
 
   componentDidMount: function() {
     let articleContainer = document.getElementById('article-container');
-    this.annotationsObject = new TextHighlighter(articleContainer);
+  },
+
+  serializeData: function() {
+    return this.props.Highlight.highlights;
```

Comment:
also, what is it doing here if it is attached to parent props? serialize should be an internal method passed to parent which triggers the redux flow.